src/main/java/org/entcore/blog/services/impl/DefaultPostService.java

```diff
@@ -95,31 +107,62 @@ public void create(String blogId, JsonObject post, UserInfos author,
 		JsonObject b = Utils.validAndGet(post, FIELDS, FIELDS);
 		if (validationError(result, b)) return;
 		b.put("sorted", now);
+
+		Future<ContentTransformerResponse> contentTransformerResponseFuture;
 		if (b.containsKey("content")) {
-			b.put("contentPlain",  StringUtils.stripHtmlTag(b.getString("content", "")));
+			final String content = b.getString("content", "");
+			b.put("contentPlain",  StringUtils.stripHtmlTag(content));
+			contentTransformerResponseFuture = contentTransformerClient
+					.transform(new ContentTransformerRequest( b.getInteger("contentVersion", 0), content));
+		} else if (b.containsKey("contentJson")) {
+			final JsonObject contentJson = b.getJsonObject("contentJson", new JsonObject());
+			contentTransformerResponseFuture = contentTransformerClient
+					.transform(new ContentTransformerRequest( b.getInteger("contentVersion", 0), contentJson));
+		} else {
+			contentTransformerResponseFuture = Future.succeededFuture();
 		}
-		plugin.setIngestJobStateAndVersion(b, IngestJobState.TO_BE_SENT, version);
-		mongo.save(POST_COLLECTION, b, MongoDbResult.validActionResultHandler(event -> {
-			if(event.isLeft()){
-				log.error("Failed to create post: ", event.left().getValue());
-				result.handle(event);
-				return;
-			}
-			//#29106 avoid fetch after save
-			final String id = event.right().getValue().getString("_id");
-			b.put("_id", id);
-			//must set id before notify
-			plugin.notifyUpsert(blogId, author, b).onComplete(e->{
-				if(e.failed()){
-					plugin.setIngestJobState(b, IngestJobState.SEND_KO);
-					log.error("Failed to notify upsert post: ", e.cause());
+		contentTransformerResponseFuture.onComplete(transformerResponse -> {
+			if (transformerResponse.failed()) {
+				log.error("Error while transforming the content", transformerResponse.cause());
+			} else {
+				if (transformerResponse.result() == null) {
+					log.debug("No content transformed.");
 				} else {
-					plugin.setIngestJobState(b, IngestJobState.SENT);
+					final ContentTransformerResponse transformerResult = transformerResponse.result();
+					if(transformerResult.getHtmlContent() != null) {
+						final String content = transformerResult.getHtmlContent();
+						b.put("content", content);
+						b.put("contentPlain",  StringUtils.stripHtmlTag(content));
```

Comment:
Pourquoi ne pas extraire ce `contentPlain` directement depuis le service de transformation ?

Le service devrait pouvoir facilement parser soit le HTML, soit le JSON, et en extraire les "text nodes" afin de récupérer le contenu "plain", affranchi de tout markup. Sans péter la mémoire de la JVM en cas de gros volume.

[Voir par exemple DOMParser](https://stackoverflow.com/questions/70848839/how-to-retrieve-text-content-from-a-html-markup-string#answer-70849133)

Et dans ce cas, `StringUtils.stripHtmlTag(content)` pourrait servir de fallback si le service de transformation échoue à parser le document.